tools/downloader/topologies.py

```diff
@@ -0,0 +1,89 @@
+import os
+import argparse
+import sys
+
+from pathlib import Path
+from argparse import Namespace
+
+import common
+import downloader
+import converter
+
+_download_dir = Path(os.getenv('OPEN_MODEL_ZOO_MODELS_DIR', Path('.').resolve() / 'models'))
+_cache_dir = Path(os.getenv('OPEN_MODEL_ZOO_CACHE_DIR', _download_dir / '.cache'))
+
+class Topology:     
+    '''
+    {}
+    Model task type: {}
+
+    License: {}
+    '''
+
+    def __init__(self, model_name, *, precision=None):
+        '''
+        Downloads target model to the folder specified in _download_dir variable.
+        If there is already downloaded model - retrieves the model from cache.
+        Creates an object which stores information about the downloaded model.
+
+        Parameters
+        ----------
+            model_name
+                Target model name.
+            precision
+                Target model precision. By default is None.
+        '''
+
+        parser = argparse.ArgumentParser()
+        args = Namespace(all=False, list=None, name=model_name, print_all=False)   
+        topology = common.load_models_from_args(parser, args)[0]
+
+        model_dir = _download_dir / topology.subdirectory
+
+        if precision is None:
+            downloader.main(['--name=' + model_name,
+                             '--output_dir=' + str(_download_dir).replace('\\', '\\\\'), 
+                             '--cache_dir=' + str(_cache_dir).replace('\\', '\\\\')])
+        else:
+            downloader.main(['--name=' + model_name, '--precisions=' + precision,
+                             '--output_dir=' + str(_download_dir).replace('\\', '\\\\'), 
+                             '--cache_dir=' + str(_cache_dir).replace('\\', '\\\\')])
```

Comment:
Code duplication may lead to the bugs. Please collect the flags and then call downloader:
```python
flags = ['--name=' + model_name,
         '--output_dir=' + str(_download_dir).replace('\\', '\\\\'), 
         '--cache_dir=' + str(_cache_dir).replace('\\', '\\\\')]
if precision:
  flags += ['--precisions=' + precision]

downloader.main(flags)
```